### DISCUSSION TITLE
Loading data from loaders in nested routes in parallel

### DISCUSSION THREAD

**Discussion by maoberlehner** [Hebilicious/form-actions-nuxt#85] [Ideas]
When it comes to loaders and actions in other frameworks like Remix and SvelteKit, they emphasize their usage alongside nested routes. They load data for the whole tree of nested routes in parallel. This makes this feature even more useful.

It seems like with `form-actions-nuxt` loaders are still triggered in parallel which I find very unfortunate.

I think adding parallel loading and streaming (https://remix.run/docs/en/main/guides/streaming#3-deferring-data-in-loaders) to `form-actions-nuxt` would be a great addition to the Nuxt ecosystem in general!

**Comment by Hebilicious** [comment]
Hi @maoberlehner,
If I'm not mistaken, we need to implement this at the Nuxt level, and then it will work directly with this module. The data fetching currently uses `useFetch` under the hood. I think the best way to get this implemented would be to open an issue in https://github.com/nuxt/nuxt for parallel loading and streaming support while fetching server data. Ideally with an example illustrating how it works currently and what should happen. I think an utility similar to remix defer could be implemented either at the Nuxt or Nitro level, which would help with streaming. Nuxt already supports streaming, but I don't think this is well documented.
